tests/lib/tools/snaps-state

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: pack-local <snap-name>"
+    echo "       install-local <snap-name> [OPTIONS]"
+    echo "       install-local-as <snap-name> <dest-name> [OPTIONS]"
+    echo "       is-classic-confinement-supported"
+    echo ""
+    echo "Available options:"
+    echo "  --devmode --jailmode --classic"
+    echo ""
+    echo "Pack and install commands save the packed snap for future uses,"
+    echo "which is reused on the following calls."
+    echo "The paths for locating the sources of the snaps to either pack or"
+    echo "install are the local path and then 'tests/lib/snaps/'"
+}
+
+pack_local() {
+    local SNAP_NAME="$1"
+    local SNAP_DIR="$TESTSLIB/snaps/${SNAP_NAME}"
+    if [ $# -gt 1 ]; then
+        SNAP_DIR="$2"
+    fi
+    local SNAP_FILE="${SNAP_DIR}/${SNAP_NAME}_1.0_all.snap"
+    # assigned in a separate step to avoid hiding a failure
+    if [ ! -f "$SNAP_FILE" ]; then
+        snap pack "$SNAP_DIR" "$SNAP_DIR" >/dev/null || return 1
+    fi
+    # echo the snap name
+    if [ -f "$SNAP_FILE" ]; then
+        echo "$SNAP_FILE"
+    else
+        find "$SNAP_DIR" -name '*.snap' | head -n1
```

Comment:
This is somewhat brittle. Can we enforce that snap directory name matches the snap name, so that we can use `"$SNAP_NAME_*_all.snap"`

Reply:
install local makes sure that the dir is the same than the name, so it should work.

Reply:
Can you adjust the find to match against `"${SNAP_NAME}_*_all.snap"` then?

Reply:
I tested that and it doesn't work on some tests because for some snaps the snap name in the snap.yaml does not match with then directory name and this is intentional. 1 example is tests/lib/snaps/test-snapd-service-v2-bad

Reply:
We may want to be strict here and adjust that test, even at the cost of not using a specific helper in that test.